aiida/manage/caching.py

```diff
@@ -163,54 +188,123 @@ def _reset_config():
-        )
+    type_check(identifier, (type(None), str))
 
     with _reset_config():
         if identifier is None:
             _CONFIG[ConfigKeys.DEFAULT.value] = False
+            _CONFIG[ConfigKeys.ENABLED.value] = []
         else:
+            _validate_identifier_pattern(identifier=identifier)
             _CONFIG[ConfigKeys.DISABLED.value].append(identifier)
-            try:
+            with suppress(ValueError):
                 _CONFIG[ConfigKeys.ENABLED.value].remove(identifier)
-            except ValueError:
-                pass
         yield
+
+
+def _match_wildcard(*, string, pattern):
+    """
+    Helper function to check whether a given name matches a pattern
+    which can contain '*' wildcards.
+    """
+    regexp = '.*'.join(re.escape(part) for part in pattern.split('*'))
+    return re.fullmatch(pattern=regexp, string=string) is not None
+
+
+def _validate_identifier_pattern(*, identifier):
+    """
+    The identifier (without wildcards) can have one of two forms:
+
+    1.  <group_name><ENTRY_POINT_STRING_SEPARATOR><tail>
+
+        where `group_name` is one of the keys in `entry_point_group_to_module_path_map`
+        and `tail` can be anything _except_ `ENTRY_POINT_STRING_SEPARATOR`.
+
+    2. a fully qualified Python name
+
+        this is a colon-separated string, where each part satisfies
+        `part.isidentifier() and not keyword.iskeyword(part)`
+
+    This function checks if an identifier _with_ wildcards can possibly
+    match one of these two forms. If it can not, a `ValueError` is raised.
+
+    :param identifier: Process type string, or a pattern with '*' wildcard that matches it.
+    :type identifier: str
+    """
+    common_error_msg = "Invalid identifier pattern '{}': ".format(identifier)
+    assert ENTRY_POINT_STRING_SEPARATOR not in '.*'  # The logic of this function depends on this
+    # Check if it can be an entry point string
+    if identifier.count(ENTRY_POINT_STRING_SEPARATOR) > 1:
+        raise ValueError(
+            common_error_msg +
+            "Can contain at most one entry point string separator '{}'".format(ENTRY_POINT_STRING_SEPARATOR)
+        )
+    # If there is one separator, it must be and entry point string.
+    # Check if the left hand side is a matching pattern
+    if ENTRY_POINT_STRING_SEPARATOR in identifier:
+        group_pattern, _ = identifier.split(ENTRY_POINT_STRING_SEPARATOR)
+        if not any(
+            _match_wildcard(string=group_name, pattern=group_pattern)
+            for group_name in entry_point_group_to_module_path_map
+        ):
+            raise ValueError(
+                common_error_msg + "Group name pattern '{}' does not match any of the AiiDA entry point group names.".
+                format(group_pattern)
+            )
+        # The group name pattern matches, and there are no further
+        # entry point string separators in the identifier, hence it is
+        # a valid pattern.
```

Comment:
I was just wondering if we should maybe still check the `tail` and see if it corresponds to an actual registered entry point, _if_ it is a fully specified entry point and if not, emit a warning. I am thinking of someone adding a fully qualified entry point string to the config but it contains a type, and then being really surprised that the corresponding class is not cached. Having the warning would help I imagine. But then we would have to cache the parsing and validation of the config as we don't want to rerun this everytime the config is retrieved. What do you think? Too much for now?

Reply:
Hmm yeah, if there aren't any wildcards we could essentially do the same validation that was there before - for me it would even be fine to make it an error (warnings are very easily ignored..).

Why would it need to be cached, though? It only happens at startup, same as before. It seems difficult to _both_ check that it corresponds to a currently-available entry point _and_ cache it between interpreter restarts.

> but it contains a type
I think you mean "typo" 😄

Reply:
> I think you mean "typo" smile

Haha, I guess one would call that a Freudian ~~slip~~ typo :wink:
Regarding the caching: I quickly looked at the code and didn't see the parsing happening only once, but if that is the case, then there is no problem.

The reason I  think raising an error is maybe not the right choice is that then one is _forced_ to have only entry points in there that are currently installed. Imagine I have a working config and then uninstall a plugin, then all startups the config will crash, which would be quite annoying. But maybe we can just leave this for now. If it ever comes up as a problem then we can always add more heavy validation.

If you can just address the one remaining change to `allow_none=True` in the `type_check` we can merge this.

Reply:
Ok, I've fixed the ``type_check`` instances.

Parsing happens only once because the ``_with_config`` decorator only parses it if ``_CONFIG`` is empty. 

We probably don't need to handle the "`*` in the entrypoint name" case, right? The only limitation is that there's no way to put a literal / escaped `*` into your configuration pattern, or it will think it's a wildcard.

Reply:
Hmm yeah, sub ideal. But I think it is ok, because even though technically allowed in an entry point name, I don't think anyone will and we can add a section to the docs and registry that people should not use it

Reply:
Well, technically, we could just replace all instances of where we use ``*`` to look for wildcards with the ``(^|[^\\])(\*)`` regex (literal `*`, preceded either by start of string or _not_ ``\``).

Might as well do things right.. but this is about as unlikely as it gets to ever be used.

Reply:
But of course, then the rabbit hole really opens up.. ``foo\\*bar`` is an escaped ``\`` followed by an _unescaped_ wildcard. There's only so much you can do before implementing a full fledged grammar. I think it's fine to leave as is.

Reply:
yeah, that way madness lies. So that means this beast is ready to go, right?

Reply:
yup, good to go for me